utils/flare.py

```diff
@@ -0,0 +1,322 @@
+        self._forwarder_log = config.get('{0}forwarder_log_file'.format(prefix))
+        self._dogstatsd_log = config.get('{0}dogstatsd_log_file'.format(prefix))
+        self._jmxfetch_log = config.get('jmxfetch_log_file')
+
+    # Add logs to the tarfile
+    def _add_logs_tar(self):
+        self._add_log_file_tar(self._collector_log)
+        self._add_log_file_tar(self._forwarder_log)
+        self._add_log_file_tar(self._dogstatsd_log)
+        self._add_log_file_tar(self._jmxfetch_log)
+        self._add_log_file_tar(
+            "{0}/*supervisord.log*".format(os.path.dirname(self._collector_log))
+        )
+
+    def _add_log_file_tar(self, file_path):
+        for f in glob.glob('{0}*'.format(file_path)):
+            log.info("  * {0}".format(f))
+            self._tar.add(
+                f,
+                os.path.join(self._prefix, 'log', os.path.basename(f))
+            )
+
+    # Collect all conf
+    def _add_conf_tar(self):
+        conf_path = get_config_path()
+        log.info("  * {0}".format(conf_path))
+        self._tar.add(
+            self._strip_comment(conf_path),
+            os.path.join(self._prefix, 'etc', 'datadog.conf')
+        )
+
+        if not Platform.is_windows():
+            supervisor_path = os.path.join(
+                os.path.dirname(get_config_path()),
+                'supervisor.conf'
+            )
+            log.info("  * {0}".format(supervisor_path))
+            self._tar.add(
+                self._strip_comment(supervisor_path),
+                os.path.join(self._prefix, 'etc', 'supervisor.conf')
+            )
+
+        for file_path in glob.glob(os.path.join(get_confd_path(), '*.yaml')):
+            self._add_clean_confd(file_path)
+
+    # Return path to a temp file without comment
+    def _strip_comment(self, file_path):
+        _, temp_path = tempfile.mkstemp(prefix='dd')
+        atexit.register(os.remove, temp_path)
+        temp_file = open(temp_path, 'w')
+        orig_file = open(file_path, 'r').read()
+
+        for line in orig_file.splitlines(True):
+            if not self.COMMENT_REGEX.match(line):
+                temp_file.write(re.sub(self.APIKEY_REGEX, self.REPLACE_APIKEY, line))
+        temp_file.close()
+
+        return temp_path
+
+    # Remove password before collecting the file
+    def _add_clean_confd(self, file_path):
+        basename = os.path.basename(file_path)
+
+        temp_path, password_found = self._strip_password(file_path)
+        log.info("  * {0}{1}".format(file_path, password_found))
+        self._tar.add(
+            temp_path,
+            os.path.join(self._prefix, 'etc', 'conf.d', basename)
+        )
+
+    # Return path to a temp file without password and comment
+    def _strip_password(self, file_path):
+        _, temp_path = tempfile.mkstemp(prefix='dd')
+        atexit.register(os.remove, temp_path)
+        temp_file = open(temp_path, 'w')
+        orig_file = open(file_path, 'r').read()
```

Comment:
It looks like you are not closing orig_file.

You should use python "with" syntax to handle files properly